docs/spec/lightclient/detection.md

```diff
@@ -0,0 +1,491 @@
+Let *b* be a light block. 
+We define *sequ-rooted(b)* iff for all i, 1 <= i < h = b.Header.Height,
+there exist light blocks a(i) s.t.
+   - *a(1) = Genesis* and
+   - *a(h) = b* and
+   - *signs( a(i) , a(i+1) )*.
+
+> The following formalizes that *c* is trusted based on *b* in
+> skipping verification. Observe that we do not require here (yet)
+> that *b* was properly generated.
+
+#### **[TMBC-SKIP-ROOT]**
+Let *b* and *c* be light blocks. We define *skip-root(b,c,t)* if at
+time t there exists an *h* and a sequence *a(1)*, ... *a(h)* s.t.
+   - *a(1) = b* and
+   - *a(h) = c* and
+   - *supports( a(i), a(i+1), t)*, for all i, *1 <= i < h*.
+
+> **TODO** In the above we might use a sequence of times t(i). Not sure.
+
+> **TODO:** I believe the following definition
+> corresponds to **Slashable fork** in
+> [forks][tendermintfork]. Please confirm!  
+> Observe that sign-skip-match  is even defined if there is a fork on
+> the chain.
+
+#### **[TMBC-SIGN-SKIP-MATCH]**
+Let *a*, *b*, *c*, be light blocks and *t* a time, we define 
+*sign-skip-match(a,b,c,t) = true* iff
+   - *sequ-rooted(a)* and
+   - *sequ-rooted(b)* and
+   - *b.Header.Height = c.Header.Height* and
+   - *skip-root(a,c,t)*
+
+implies *b = c*.
+
+If there exists three light blocks a, b, and c, with 
+*sign-skip-match(a,b,c,t) =
+false* then we have a *slashable fork*.
+
+
+
+> **TODO:** I think the following definition is
+> the intuition behind **main chain forks**
+> in the document on [forks][tendermintfork]. However, main chain
+> forks were defined more operational "forks that are observed by
+> full nodes as part of normal Tendermint consensus protocol". Please
+> confirm! 
+  
+#### **[TMBC-SIGN-UNIQUE]**
+Let *b* and *c* be  light blocks, we define *sign-unique(b,c) =
+true* iff
+   - *b.Header.Height =  c.Header.Height* and
+   - *sequ-rooted(b)* and
+   - *sequ-rooted(c)*
+   
+implies *b = c*.
+
+If there exists two light blocks b and c, with *sign-unique(b,c) =
+false* then we have a *fork on the chain*.
+
+> The following captures what I believe is called a light client fork
+> in our discussions. There is no fork on the chain up to the height 
+> of block b. However, c is of that height, is different, and passes skipping
+> verification  
+> Observe that a light client fork is a special case of a slashable
+> fork. 
+
+
+#### **[TMBC-LC-FORK]**
+Let *a*, *b*, *c*, be light blocks and *t* a time. We define
+*light-client-fork(a,b,c,t)* iff
+   - *sign-skip-match(a,b,c,t) = false* and
+   - *sequ-rooted(b)* and
+   - *b* is "unique", that is, for all *d*,  *sequ-rooted(d)* and
+     *d.Header.Height=b.Header.Height* implies *d = b*
+
+
+> Finally, let's also define bogus blocks that have no support.
+> Observe that bogus is even defined if there is a fork on the chain.
+
+#### **[TMBC-BOGUS]**
+Let *b* be a light block and *t* a time. We define *bogus(b,t)* iff
+  - *sequ-rooted(b) = false* and
+  - for all *a*, *sequ-rooted(a)* implies *skip-root(a,b,t) = false*
```

Comment:
Should a.height be smaller than b.height here?

Reply:
yes. it is implied by the definitions. but I will add a remark for clarificiation.